### PR TITLE
[export][refactor][2/n] Move tracing logic

### PR DESCRIPTION
**Pull request by angelayi**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #114787
* __->__ #114768
* #114764

2/n of refactoring export code:

* Moved tracing logic in torch/_export/init.py to torch/export/_tracer.py

Differential Revision: [D51823961](https://our.internmc.facebook.com/intern/diff/D51823961)

### CODE REVIEW

File: torch/_export/__init__.py [hunk 1]
```diff
@@ -229,90 +219,6 @@ def _eval(self, mode: bool = True):
...
-                    except Exception:  # TODO(zhxchen17) Remove this.
-                        return path
-
-                nn_module_stack = {root_key: (root, root_cls), **nn_module_stack}
-                node.meta["nn_module_stack"] = {
-                    key: (normalize_path(path), ty)
-                    for key, (path, ty) in nn_module_stack.items()
-                }
-
 def _export_to_torch_ir(
```
Comment (ydwu4) [inline torch/_export/__init__.py:222]: why we want to keep this function and reforward to _trace._export_to_torch_ir? Can we just do from torch.export._trace import _export_to_torch_ir and remove this forwarding?

File: torch/_export/__init__.py [hunk 2]
```diff
@@ -588,262 +381,17 @@ def _export(
...
-            {},
-            [],
-            [ModuleCallEntry("", ModuleCallSignature([], [], pytree.tree_flatten((args, {}))[1], out_spec))],
-            (args, kwargs),
-            tensor_constants=ep_non_strict.tensor_constants,
-        )
-
+    from torch.export._trace import _export
 
-    gm_torch_level = _export_to_torch_ir(
```
Comment (ydwu4) [inline torch/_export/__init__.py:640]: Maybe also remove this forwarding if we can.

**Review by ydwu4 — APPROVED** [review]

File: torch/_export/__init__.py (hunk 1 above)
Comment (angelayi) [reply torch/_export/__init__.py:222]: Eventually I want to get rid of reliance on torch/_export, and move all calls to torch.export._trace.export_to_torch_ir. Might as well remove it in this diff and do the forwarding in torch._export. Does that make sense?

**Comment by angelayi** [comment]
@angelayi has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D51823961).

**Comment by angelayi** [comment]
@angelayi has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D51823961).

**Pull request closed without merging**